slack_sdk/socket_mode/aiohttp/__init__.py

```diff
@@ -253,12 +252,19 @@ async def receive_messages(self) -> None:
                 self.logger.debug("The running receive_messages task is now cancelled")
             raise
 
+    async def is_ping_pong_failing(self) -> bool:
+        if self.last_ping_pong_time is None:
+            return False
+        disconnected_seconds = int(time.time() - self.last_ping_pong_time)
+        return disconnected_seconds >= (self.ping_interval * 4)
+
     async def is_connected(self) -> bool:
         return (
             not self.closed
             and not self.stale
             and self.current_session is not None
             and not self.current_session.closed
+            and not await self.is_ping_pong_failing()
```

Comment:
Moved the ping-pong validation logic to is_ping_pong_failing method and added it to is_connected() method